tests/transforms/test_draw_mpl.py

```diff
@@ -0,0 +1,196 @@
+    assert len(ax.lines) == 6  # three wires, three lines for CNOT
+    assert len(ax.texts) == 5  # three wire labels, 2 box labels
+
+    assert ax.texts[0].get_text() == "0"
+    assert ax.texts[1].get_text() == "a"
+    assert ax.texts[2].get_text() == "1.23"
+
+    # gates in same layer can be in any order
+
+    texts = [t.get_text() for t in ax.texts[3:]]
+    assert "RX" in texts
+    assert "RY" in texts
+    plt.close()
+
+
+def test_decimals():
+    """Test decimals changes operation labelling"""
+
+    _, ax = qml.draw_mpl(circuit1, decimals=2)(1.23, 2.34)
+
+    texts = [t.get_text() for t in ax.texts[3:]]
+    assert "RX\n(1.23)" in texts
+    assert "RY\n(2.34)" in texts
+    plt.close()
+
+
+def test_label_options():
+    """Test label options modifies label style."""
+
+    _, ax = qml.draw_mpl(circuit1, label_options={"color": "purple", "fontsize": 20})(1.23, 2.34)
+
+    for l in ax.texts[0:3]:  # three labels
+        assert l.get_color() == "purple"
+        assert l.get_fontsize() == 20
+    plt.close()
+
+
+class TestWireBehaviour:
+    """Tests that involve how wires are displayed"""
+
+    def test_wire_order(self):
+        """Test wire_order changes order of wires"""
+
+        _, ax = qml.draw_mpl(circuit1, wire_order=(1.23, "a"))(1.23, 2.34)
+
+        assert len(ax.texts) == 5
+
+        assert ax.texts[0].get_text() == "1.23"
+        assert ax.texts[1].get_text() == "a"
+        assert ax.texts[2].get_text() == "0"
+        plt.close()
+
+    def test_empty_wires(self):
+        """Test empty wires do not appear by default"""
+
+        _, ax = qml.draw_mpl(circuit2)(1.23)
+
+        assert len(ax.lines) == 1  # one wire
+        assert len(ax.texts) == 2  # one wire label and one gate label
+        assert ax.texts[0].get_text() == "0"
+        assert ax.texts[1].get_text() == "RX"
+        plt.close()
+
+    def test_show_all_wires(self):
+        """Test show_all_wires=True displays empty wires."""
+
+        _, ax = qml.draw_mpl(circuit2, show_all_wires=True)(1.23)
+
+        assert len(ax.lines) == 3  # three wires
+
+        assert len(ax.texts) == 4  # three wire labels and one gate label
+        assert ax.texts[0].get_text() == "0"
+        assert ax.texts[1].get_text() == "a"
+        assert ax.texts[2].get_text() == "1.23"
+        plt.close()
+
+    def test_wire_order_not_on_device(self):
+        """Test when ``wire_order`` priority by requesting ``show_all_wires`` with
+        the ``wire_order`` containing wires not on the device. The output should have
+        three wires, one active and two empty wires from the wire order."""
+
+        _, ax = qml.draw_mpl(circuit2, wire_order=[2, "a"], show_all_wires=True)(1.23)
+
+        assert len(ax.lines) == 3  # three wires
+
+        assert len(ax.texts) == 4  # three wire labels and one gate label
+        assert ax.texts[0].get_text() == "2"
+        assert ax.texts[1].get_text() == "a"
```

Comment:
Just a random thought, maybe as a separate feature, could we assign a single label to a group of wires, lets say for example we have 5 wires, 2 are ancillary and the remaining 3 constitute a state $\psi$